### DISCUSSION TITLE
Tailwind missing from default layout

### DISCUSSION THREAD

**Discussion by bergstar** [livewire/livewire#9056] [7. Bugs (archived)]
### Livewire version

v3.5.15

### Laravel version

v11.34.2

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Follow steps on https://livewire.laravel.com/docs/quickstart

Get to the step:
_"You may have noticed there is no JavaScript or CSS assets provided by Livewire. That is because Livewire 3 and above automatically injects any frontend assets it needs."_

Try out any Tailwind stuff and confirm that it indeed isnt working. 

Tested several times.

### Repository or Wirebox URL

https://livewire.laravel.com/docs/quickstart

### Additional information

I think its essential that documentation is easy to use for new developers and produces consistent results.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by bergstar** [comment]
Indeed it needs:
```
    <!-- Styles / Scripts -->
    @if (file_exists(public_path('build/manifest.json')) || file_exists(public_path('hot')))
        @vite(['resources/css/app.css', 'resources/js/app.js'])
    @endif

    @livewireStyles

...

@livewireScripts
```

This is despite the documentation saying
`By default, Livewire injects the JavaScript and CSS assets it needs into each page that includes a Livewire component.
`

**Comment by joshhanley** [comment]
@bergstar thanks for reporting! You are correct Livewire doesn't inject Tailwind. That's because it doesn't depend on Tailwind, so it will only "injects any frontend assets **IT** needs". Meaning it only injects Livewire's style and Livewire's javascript. You still need to add Tailwind, or Bootstrap, or whatever styling library you are using to your layout and any custom javascript if you have any.

**Reply by bergstar** [reply]
No, that is not correct. Tailwind now ships with Laravel as standard. Installing Livewire breaks that. 
`php artisan livewire:layout` as a bare minimum should keep functionality of welcome.blade.php that Laravel ships with.  

Not only that, but livewire it self isnt functioning properly without @livewireStyles / @livewireScripts - both are not present in default layout.

Either documentation should be updated to reflect that default installation process isnt complete and add steps person should take manually. Or just fix the layout file.

Regardless, currently in order to get a working boilerplate new users must buy a 100$ screencast to get started properly.

**Reply by joshhanley** [reply]
Hi @bergstar, I’d be happy to help address any concerns you have, but I want to point out that the tone of your message comes across as unnecessarily confrontational. Let’s keep this conversation constructive.

To clarify - my initial message was accurate. Livewire doesn’t depend on Tailwind, so it doesn’t manage Tailwind’s inclusion, Livewire's auto injection focuses on injecting only the assets it needs to function.

I didn't say that Laravel doesn't include Tailwind so please don't misconstrue my words. As it turns out, it's only a relatively recent change that Laravel started including Vite assets by default as an addition to including the whole Tailwind CSS compiled file in the welcome view and providing a basic Tailwind configuration out of the box https://github.com/laravel/laravel/pull/6463.

As this is an open source project, PRs are always welcome to update the project to improve it or to bring it into line with new changes in Laravel. PRs are also welcome to improve the documentation.

Regarding your issue that Livewire's `@livewireStyles / @livewireScripts` aren't included in the layout, this is because in V3 the styles and scripts assets are auto-injected, so we no longer need to manually add them to the layout file. If you installation isn't working, then please open a separate issue we a demo repo showing that asset auto-injection not working so we can investigate it properly.

Finally, I'm not sure why you believe users need to purchase the screencasts to get started properly as the whole "Getting Started" section of the screencasts is free to access. If something on our site has misled you'd, please show us and we'd be happy to look into it.

**Reply by bergstar** [reply]
Let's boil it down to the basics - regarding '@livewireStyles / @livewireScripts' its a Livewire thing. There should be some clarity in the documentation and installation - but in the end of the day it's a Livewire thing.

Package breaking existing functionality of Laravel is the important part here. I also cant understand why spend all that energy discussing something that is latterly 1 copy-paste away from fixing? 

And yes, Im very annoyed with the way this project manages its documentation - watching the whole screencast is essential to get started with Livewire properly. Im strongly against this kind of informational gatekeeping. Its not even part of laracasts and clearly overpriced.

**Reply by lrljoe** [reply]
Hey @bergstar!

The docs are pretty clear in terms of Installation/Getting Started.

There are sections in the docs that explain about layouts, assets etc, and there are millions of sites out there using Livewire, and very few of the devs for those have purchased any screencasts etc.

As has been explained above by one of the core maintainers, Livewire is agnostic in terms of which CSS framework (or not) you choose to use.  
This is why it doesn't include references to Tailwind etc.  Adding references to these would make them dependencies, (when they're not actually needed!), and this would subsequently end up actually breaking more than it fixes. 

This would then prove confusing to anyone who's not using Tailwind (for example), as they'd think it was needed (when it isn't).  It is your choice to use Tailwind, the same way as it's someone else's choice to use Bootstrap instead.

You're the first person I've seen wanting this, and with 37,803,989 installs, it's clearly been understood by everyone else.

Joe.

**Reply by samlev** [reply]
Livewire isn't breaking anything - it's providing the absolute, minimal app layout _if you don't already have one_. You aren't expected to use it unchanged, it's just a blank starting point if you don't have anything else already.

You're _meant_ to modify it for your use case, which includes adding in `@vite()` if you need to.

**Comment by bergstar** [comment]
"Livewire is a full-stack framework for Laravel". Since it is a full-stack framework for Laravel - it should be provided with frontend and backend base template which is actually working. Currently, with current template, without Tailwind, @livewireStyles / @livewireScripts it is not usable as a full-stack framework after a basic installation. 

Livewire cant be used without Laravel. And Laravel has made a decision to bundle Tailwind. Livewire should respect that discussion. 

Package should be possible to use out of the box after intallation without getting a "starter kit", watching screencasts etc. 

I dont understand why there is such a strong will to gatekeeping a framework from new users ?
Or is it again about the money and making it more user friendly for @fluxStyles / @fluxScripts ?

**Reply by samlev** [reply]
It's not gatekeeping. It's a single line that you _might_ have to add to a file that you will definitely have to change anyway, and which is provided for you to use only if you don't already have anything. It's a stub file.

**Reply by lrljoe** [reply]
In short, sorry, but no.  Livewire is agnostic by design.

The stub file is just there to give you something to start with, that will function, out of the box, without further work being needed.

It's then up to you to decide how to build your ux/frontend.

Most users will follow the installation guide, and sensibly migrate existing layouts etc

There's no gatekeeping involved at all, the docs are very comprehensive.  Anyone starting out with Livewire would need to read them to understand how it all works anyway.

Just because YOU choose to use Tailwind, doesn't mean that all packages must abide by what you do, and defaulting to it would be more confusing to new users, as they'd assume that tailwind is required for Livewire to work.

I'm not getting paid for contributions to Livewire, nor are the vast majority of the community, before you keep insinuating that it's all financially motivated.
